src/lib/services/suggest-change-service.ts

```diff
@@ -0,0 +1,62 @@
+import { IUnleashConfig, IUnleashStores } from '../types';
+import { Logger } from '../logger';
+import { IEventStore } from '../types/stores/event-store';
+import { SUGGEST_CHANGE_CREATED } from '../types/events';
+import User from '../types/user';
+import { ISuggestChangeStore } from '../types/stores/suggest-change-store';
+import { ISuggestChangeSet } from '../types/model';
+
+export default class SuggestChangeService {
+    private config: IUnleashConfig;
+
+    private logger: Logger;
+
+    private suggestChangeStore: ISuggestChangeStore;
+
+    private eventStore: IEventStore;
+
+    constructor(
+        {
+            suggestChangeStore,
+            eventStore,
+        }: Pick<IUnleashStores, 'suggestChangeStore' | 'eventStore'>,
+        config: IUnleashConfig,
+    ) {
+        this.config = config;
+        this.logger = config.getLogger('services/suggest-change-service.ts');
+        this.suggestChangeStore = suggestChangeStore;
+        this.eventStore = eventStore;
+    }
+
+    async createChangeRequest(
```

Comment:
Do we need a method to record events after it has been created? Does creating a changeRequest always require a changeset?

Reply:
A change request requires a changeSet, yes